### PR TITLE
Add support for resources embedded in KMZ files

### PR DESCRIPTION
**Pull request by balhar-jakub**
### Description of the Change

Update resolution of links in the KML Files to allow resources packaged in the KMZ packages. 

### Why Should This Be In Core?

Plenty of KMZ files use internal resources and therefore we should support them. 

### Benefits

The resources packaged in the KMZ file will be correctly interpreted. 

### Applicable Issues

### CODE REVIEW

File: src/formats/kml/util/StyleResolver.js [hunk 1]
```diff
@@ -34,58 +36,75 @@ define([
...
-    // Intentionally undocumented. For internal use only
-    StyleResolver.prototype.handleStyleSelector = function (styleSelector, resolve, reject) {
+    /**
+     * It handles style selector. Either by resolving a map or by simply returning the resolved promise with style data.
+     * @param styleSelector
+     * @returns {Promise|*}
+     * @private
+     */
+    StyleResolver.prototype.handleStyleSelector = function (styleSelector) {
+        console.log(styleSelector);
```
Comment (AkeluX) [inline src/formats/kml/util/StyleResolver.js]: Is this needed?

File: src/formats/kml/styles/KmlStyleMap.js [hunk 2]
```diff
@@ -95,7 +91,8 @@ define([
                 results['highlight'] =  null;
             }
 
-            resolve(results);
+            console.log(results);
```
Comment (AkeluX) [inline src/formats/kml/styles/KmlStyleMap.js]: Is this needed?

File: src/formats/kml/KmzFile.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
```
Comment (AkeluX) [inline src/formats/kml/KmzFile.js]: Please update to the new file header format.

File: src/formats/kml/util/HrefResolver.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+define([
```
Comment (AkeluX) [inline src/formats/kml/util/HrefResolver.js]: Please add the standard file header.

File: src/formats/kml/util/StyleResolver.js (hunk 1 above)
Comment (balhar-jakub) [reply src/formats/kml/util/StyleResolver.js]: No

File: src/formats/kml/styles/KmlStyleMap.js (hunk 2 above)
Comment (balhar-jakub) [reply src/formats/kml/styles/KmlStyleMap.js]: No

File: src/formats/kml/KmzFile.js (hunk 3 above)
Comment (balhar-jakub) [reply src/formats/kml/KmzFile.js]: Done

File: src/formats/kml/util/HrefResolver.js (hunk 4 above)
Comment (balhar-jakub) [reply src/formats/kml/util/HrefResolver.js]: Done

**Pull request merged**